File: Tutoriels/Python/POO/Heritage/src/GPersonStudent.py
#================================================
from GPerson import GPerson
#================================================
class GPersonStudent(GPerson): 
    def __init__(self, name, average):
        GPerson.__init__(self, name)
        self.m_average = average
    def __str__(self):
        return "Etudiant: {0} | {1}".format(GPerson.__str__(self), self.m_average)
#================================================

File: Tutoriels/Cpp/POO/Classe_Abstraite/src/GPersonStudent.cpp
//===============================================
#include "GPersonStudent.h"
//===============================================
GPersonStudent::GPersonStudent() {
    m_average = 0.0;
}
//===============================================
GPersonStudent::GPersonStudent(const string& name, const string& email, const string& average) :
    GPerson(name, email) {
    m_average = average;
}
//===============================================
GPersonStudent::~GPersonStudent() {

}
//===============================================
void GPersonStudent::print() const {
    string lData = "";
    lData += "Etudiant: ";
    lData += "Nom: " + m_name + " | ";
    lData += "Email: " + m_email + " | ";
    lData += "Moyenne: " + m_average;
    lData += "\n";
    cout << lData.c_str();
}
//===============================================
void GPersonStudent::printType() const {
    string lData = "";
    lData += "Etudiant: Je suis Etudiant et ma moyenne est de ";
    lData += m_average;
    lData += "\n";
    cout << lData.c_str();
}
//===============================================
